flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/benchmark/SingleInputGateBenchmarkFactory.java

```diff
@@ -0,0 +1,173 @@
+	protected InputChannel createKnownInputChannel(
+			SingleInputGate inputGate,
+			int index,
+			NettyShuffleDescriptor inputChannelDescriptor,
+			SingleInputGateFactory.ChannelStatistics channelStatistics,
+			InputChannelMetrics metrics) {
+		ResultPartitionID partitionId = inputChannelDescriptor.getResultPartitionID();
+		if (inputChannelDescriptor.isLocalTo(taskExecutorResourceId)) {
+			return new TestLocalInputChannel(
+				inputGate,
+				index,
+				partitionId,
+				partitionManager,
+				taskEventPublisher,
+				partitionRequestInitialBackoff,
+				partitionRequestMaxBackoff,
+				metrics);
+		} else {
+			return new TestRemoteInputChannel(
+				inputGate,
+				index,
+				partitionId,
+				inputChannelDescriptor.getConnectionId(),
+				connectionManager,
+				partitionRequestInitialBackoff,
+				partitionRequestMaxBackoff,
+				metrics,
+				networkBufferPool);
+		}
+	}
+
+	/**
+	 * A {@link LocalInputChannel} which ignores the given subpartition index and uses channel index
+	 * instead when requesting subpartition.
+	 */
+	private static class TestLocalInputChannel extends LocalInputChannel {
+
+		private ResultPartitionID newPartitionID = new ResultPartitionID();
+
+		public TestLocalInputChannel(
+				SingleInputGate inputGate,
+				int channelIndex,
+				ResultPartitionID partitionId,
+				ResultPartitionManager partitionManager,
+				TaskEventPublisher taskEventPublisher,
+				int initialBackoff,
+				int maxBackoff,
+				InputChannelMetrics metrics) {
+			super(
+				inputGate,
+				channelIndex,
+				partitionId,
+				partitionManager,
+				taskEventPublisher,
+				initialBackoff,
+				maxBackoff,
+				metrics);
+		}
+
+		@Override
+		public void requestSubpartition(int subpartitionIndex) throws IOException, InterruptedException {
+			super.requestSubpartition(channelIndex);
+		}
+
+		@Override
+		public ResultPartitionID getPartitionId() {
+			return newPartitionID;
+		}
+	}
+
+	/**
+	 * A {@link RemoteInputChannel} which ignores the given subpartition index and uses channel index
+	 * instead when requesting subpartition.
+	 */
+	private static class TestRemoteInputChannel extends RemoteInputChannel {
+
+		private ResultPartitionID newPartitionID = new ResultPartitionID();
+
+		public TestRemoteInputChannel(
+				SingleInputGate inputGate,
+				int channelIndex,
+				ResultPartitionID partitionId,
+				ConnectionID connectionId,
+				ConnectionManager connectionManager,
+				int initialBackOff,
+				int maxBackoff,
+				InputChannelMetrics metrics,
+				MemorySegmentProvider memorySegmentProvider) {
+			super(
+				inputGate,
+				channelIndex,
+				partitionId,
+				connectionId,
+				connectionManager,
+				initialBackOff,
+				maxBackoff,
+				metrics,
+				memorySegmentProvider);
+		}
+
+		@Override
+		public void requestSubpartition(int subpartitionIndex) throws IOException, InterruptedException {
+			super.requestSubpartition(channelIndex);
+		}
+
+		@Override
+		public ResultPartitionID getPartitionId() {
```

Comment:
nit: it is better to give some explanations why we need to override this method, otherwise it is not very easy understood by others.

Reply:
Ignore this because of [comment](https://github.com/apache/flink/pull/11155#discussion_r383090260)